Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -459,36 +459,56 @@ void AppendPipeline::didReceiveInitializationSegment()
             trackIndex++;
         }
     } else {
-        // Since we don't rely on the demuxer pad-added signal and this pipeline is not
-        // stream-aware, we need to account for stream topology changes ourselves.
-        unsigned videoPadsCount = 0;
-        unsigned audioPadsCount = 0;
-        unsigned textPadsCount = 0;
+        HashSet<String> videoPadStreamIDs;
+        HashSet<String> audioPadStreamIDs;
+        HashSet<String> textPadStreamIDs;
         for (auto pad : GstIteratorAdaptor<GstPad>(GUniquePtr<GstIterator>(gst_element_iterate_src_pads(m_demux.get())))) {
-            if (gst_pad_is_linked(pad))
-                continue;
             auto [parsedCaps, streamType, presentationSize] = parseDemuxerSrcPadCaps(adoptGRef(gst_pad_get_current_caps(pad)).get());
+            UNUSED_VARIABLE(parsedCaps);
+            UNUSED_VARIABLE(presentationSize);
+            GUniquePtr<char> streamIDAsCharacters(gst_pad_get_stream_id(pad));
+            String streamID = String::fromLatin1(streamIDAsCharacters.get());
```

Comment:
You don't need two variables.

    String streamID = String::fromLatin1(GUniquePtr<char>(gst_pad_get_stream_id(pad)).get());

> All temporary objects are destroyed as the last step in evaluating the [full-expression](https://en.cppreference.com/w/cpp/language/expressions#Full-expressions) that (lexically) contains the point where they were created, and if multiple temporary objects were created, they are destroyed in the order opposite to the order of creation. This is true even if that evaluation ends in throwing an exception.

Reply:
Solved